imgui_widgets.cpp

```diff
@@ -3605,14 +3603,14 @@ namespace ImStb
 
 static int     STB_TEXTEDIT_STRINGLEN(const ImGuiInputTextState* obj)                             { return obj->CurLenW; }
 static ImWchar STB_TEXTEDIT_GETCHAR(const ImGuiInputTextState* obj, int idx)                      { return obj->TextW[idx]; }
-static float   STB_TEXTEDIT_GETWIDTH(ImGuiInputTextState* obj, int line_start_idx, int char_idx)  { ImWchar c = obj->TextW[line_start_idx + char_idx]; if (c == '\n') return STB_TEXTEDIT_GETWIDTH_NEWLINE; ImGuiContext& g = *GImGui; return g.Font->GetCharAdvance(c) * (g.FontSize / g.Font->FontSize); }
+static float   STB_TEXTEDIT_GETWIDTH(ImFont* font, float fontSize, ImGuiInputTextState* obj, int line_start_idx, int char_idx)  { ImWchar c = obj->TextW[line_start_idx + char_idx]; if (c == '\n') return STB_TEXTEDIT_GETWIDTH_NEWLINE; return font->GetCharAdvance(c) * (fontSize / font->FontSize); }
 static int     STB_TEXTEDIT_KEYTOTEXT(int key)                                                    { return key >= 0x200000 ? 0 : key; }
 static ImWchar STB_TEXTEDIT_NEWLINE = '\n';
-static void    STB_TEXTEDIT_LAYOUTROW(StbTexteditRow* r, ImGuiInputTextState* obj, int line_start_idx)
+static void    STB_TEXTEDIT_LAYOUTROW(ImFont* font, float fontSize, StbTexteditRow* r, ImGuiInputTextState* obj, int line_start_idx)
 {
```

Comment:
Feels like adding `ImGuiContext*` inside `ImGuiInputTextState` would be better here?

Reply:
@ocornut 
I agree, it would prevent numerous signature change. I felt I could not take this decision myself because having `ImGuiInputTextState` pointing to an `ImGuiContext*` feels conceptually wrong regarding my interpretation of `ImGuiInputTextState`. But if you are comfortable with that choice, it definitely makes things easier.